### PR TITLE
Updates following product team review

### PR DESCRIPTION
**Pull request by agrojean-ledger**
Following comments on B2CA-953
Update Stax icons to 4bpp

### CODE REVIEW

File: src/ui/ui_main_menu_nbgl.c [hunk 1]
```diff
@@ -36,7 +36,7 @@ static bool settingsNavCallback(uint8_t page, nbgl_pageContent_t *content);
 
 #define NB_INFO_FIELDS 2
 static const char *const infoTypes[] = {"Version", "Decred App"};
```
Comment (tdejoigny-ledger) [inline src/ui/ui_main_menu_nbgl.c]: as per design guidelines, you should replace `Decred App` by `Developer` and `(c) 2023 Ledger ` by `Ledger`

File: src/ui/ui_main_menu_nbgl.c [hunk 2]
```diff
@@ -54,11 +59,6 @@ static bool settingsNavCallback(uint8_t page, nbgl_pageContent_t *content) {
         content->type = SWITCHES_LIST;
         content->switchesList.nbSwitches = 1;
         content->switchesList.switches = (nbgl_layoutSwitch_t *) &setting_switch;
```
Comment (tdejoigny-ledger) [inline src/ui/ui_main_menu_nbgl.c:61]: the setting switch  value is not restored from its previous value

File: src/ui/ui_pubkey_nbgl.c [hunk 3]
```diff
@@ -43,7 +43,7 @@ static void confirmationChoiceClbk(bool confirm) {
     switch (display_type) {
         case DISPLAY_ADDRESS:
             if (confirm) {
-                strncpy(confirm_text, "ADDRESS\nAPPROVED", sizeof(confirm_text));
+                strncpy(confirm_text, "ADDRESS\nVERIFIED", sizeof(confirm_text));
             } else {
                 strncpy(confirm_text, "Address rejected", sizeof(confirm_text));
```
Comment (tdejoigny-ledger) [inline src/ui/ui_pubkey_nbgl.c]: as per design guideline, you should replace `Address rejected` by `Address verification\ncancelled`

File: src/ui/ui_pubkey_nbgl.c [hunk 4]
```diff
@@ -70,12 +70,26 @@ static void confirmationChoiceClbk(bool confirm) {
     nbgl_useCaseStatus(confirm_text, confirm, ui_idle);
 }
 
+static void address_verification_cancelled(void) {
+    io_seproxyhal_touch_display_cancel(NULL);
+    nbgl_useCaseStatus("Address rejected", false, ui_idle);
```
Comment (tdejoigny-ledger) [inline src/ui/ui_pubkey_nbgl.c]: as per design guideline, you should replace `Address rejected` by `Address verification\ncancelled`

File: src/ui/ui_pubkey_nbgl.c [hunk 5]
```diff
@@ -70,12 +70,26 @@ static void confirmationChoiceClbk(bool confirm) {
...
+}
+
+static void ui_display_addr(void) {
+    nbgl_useCaseAddressConfirmation((char*) G_io_apdu_buffer + 200, &confirmationChoiceClbk);
+}
+
 static void warningChoiceClbk(bool reject) {
     if (reject) {
         io_seproxyhal_touch_display_cancel(NULL);
         nbgl_useCaseStatus("Address rejected", false, ui_idle);
```
Comment (tdejoigny-ledger) [inline src/ui/ui_pubkey_nbgl.c]: as per design guideline, you should replace `Address rejected` by `Address verification\ncancelled`

File: test/test_show_address.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
+from ragger.navigator import NavInsID
```
Comment (tdejoigny-ledger) [inline test/test_show_address.py:1]: reject flows not tested for both usual and unusual derviation path
QR code display not tested

File: test/test_show_address.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            navigator.navigate_until_text_and_compare(
+                NavInsID.TAPPABLE_CENTER_TAP, [
+                    NavInsID.USE_CASE_ADDRESS_CONFIRMATION_TAP,
+                    NavInsID.WAIT_FOR_HOME_SCREEN
+                ], "Show",
+                Path(__file__).parent.resolve(), path)
+        else:
+            navigator.navigate_until_text_and_compare(
+                NavInsID.RIGHT_CLICK, [NavInsID.BOTH_CLICK], "Approve",
+                Path(__file__).parent.resolve(), path)
```
Comment (tdejoigny-ledger) [inline test/test_show_address.py]: could be good to replace `Path(__file__).parent.resolve(), path)` by something like `ROOT_SCREENSHOT_PATH `with `ROOT_SCREENSHOT_PATH = Path(__file__).parent.resolve()` defined at top level

File: src/ui/ui_main_menu_nbgl.c (hunk 1 above)
Comment (agrojean-ledger) [reply src/ui/ui_main_menu_nbgl.c]: Done

File: src/ui/ui_main_menu_nbgl.c (hunk 2 above)
Comment (agrojean-ledger) [reply src/ui/ui_main_menu_nbgl.c:61]: It is on line 58 : 
`setting_switch.initState = (bool) !N_btchip.pubKeyRequestRestriction;` or maybe you mean something else

File: src/ui/ui_pubkey_nbgl.c (hunk 5 above)
Comment (agrojean-ledger) [reply src/ui/ui_pubkey_nbgl.c]: Ok

File: src/ui/ui_pubkey_nbgl.c (hunk 4 above)
Comment (agrojean-ledger) [reply src/ui/ui_pubkey_nbgl.c]: Ok

File: src/ui/ui_pubkey_nbgl.c (hunk 3 above)
Comment (agrojean-ledger) [reply src/ui/ui_pubkey_nbgl.c]: Ok

File: test/test_show_address.py (hunk 7 above)
Comment (agrojean-ledger) [reply test/test_show_address.py]: Done

File: src/ui/ui_main_menu_nbgl.c (hunk 2 above)
Comment (tdejoigny-ledger) [reply src/ui/ui_main_menu_nbgl.c:61]: not sure to understand why you need a `!` here

**Review by tdejoigny-ledger — APPROVED** [review]

File: test/test_show_address.py (hunk 6 above)
Comment (agrojean-ledger) [reply test/test_show_address.py:1]: Done.

**Pull request merged**